media/CircleEditor/view-sidebar.js

```diff
@@ -538,37 +603,261 @@ class NodeAttributeView {
                 this._element.appendChild(valueLine);
             }
         } else {
-            this._expander.innerText = '+';
-            while (this._element.childElementCount > 2) {
+            this._expander.className =
+                'sidebar-view-item-value-expander codicon codicon-chevron-down';
+            while (this._element.childElementCount > 3) {
                 this._element.removeChild(this._element.lastChild);
             }
         }
     }
 
-    /**
-     * TODO Implement edit()
-     * Attributes edit function
-     */
+    edit() {
+        while (this._element.childElementCount) {
+            this._element.removeChild(this._element.lastChild);
+        }
+        const type = this._attribute.type;
 
-    /**
-     * TODO Implement save()
-     * Attributes save function
-     */
+        if (this._isCustom === true) {
+            const input = this._host.document.getElementById('attribute' + this._index);
+            input.disabled = false;
+            this._attributeName = input.value;
+        }
+
+        this._save = this._host.document.createElement('div');
+        this._cancel = this._host.document.createElement('div');
+        this._remove = this._host.document.createElement('div');
+        this._save.className = 'sidebar-view-item-value-save codicon codicon-save';
+        this._cancel.className = 'sidebar-view-item-value-cancel codicon codicon-discard';
+        this._remove.className = 'sidebar-view-item-value-remove codicon codicon-trash';
+        this._save.addEventListener('click', (e) => {
+            e.preventDefault();
+            this.save();
+        });
+        this._cancel.addEventListener('click', () => {
+            this.cancel();
+        });
+        this._remove.addEventListener('click', () => {
+            this.remove();
+        });
+        if (this._isCustom === true) {
+            this._element.appendChild(this._remove);
+        }
+        this._element.appendChild(this._cancel);
+        this._element.appendChild(this._save);
 
-    /**
-     * TODO Implement cancel()
-     * Attributes cancel function
-     */
+        const value = this._attribute.value;
 
-    /**
-     * TODO Implement remove()
-     * When in custom operator, attributes remove function
-     */
+        switch (type) {
```

Comment:
What is this switch-case statement for? @.@
I am hard to distinguish the purpose of each case...

Reply:
This code was already implemented in Netron, so I used it.

Reply:
Thanks for the information :)
But it is my first time to see these codes and wonder what the purpose is...!
You might use these codes because you know the purpose of these codes, isn't it?